### PR TITLE
Fix non-threadsafe autoloading of tilt/haml,sass

### PR DESCRIPTION
**Pull request by XenHat**
Fixes the following warnings:
WARN: tilt autoloading 'tilt/haml' in a non thread-safe way; explicit require 'tilt/haml' suggested.
WARN: tilt autoloading 'tilt/sass' in a non thread-safe way; explicit require 'tilt/sass' suggested.

### CODE REVIEW

**Comment by XenHat** [comment]
Prod.

**Comment by charles-hollenbeck** [comment]
@XenHat Change does more than just fix haml/sass. Can you please summarize the changes on https://github.com/0xC7/That-one-file-viewer/pull/7/commits/c921269c115fc85ff6e37625a2af849837845337 ?

**Comment by XenHat** [comment]
c921269c115fc85ff6e37625a2af849837845337 profides metadata to software able to show an url preview, such as Telegram, and in theory others.

File: sharex_server.rb [hunk 1]
```diff
@@ -74,6 +105,23 @@
   send_file "views/assets/swf/#{params[:flash]}.swf"
 end
 
+# Handle POST-request (Receive and save the uploaded file)
```
Comment (XenHat) [inline sharex_server.rb]: This is an HTTP uploader.

File: views/layout.haml [hunk 2]
```diff
@@ -3,6 +3,8 @@
 	%head
 		%title= yield_content :title
 		%meta(charset="UTF-8")/
+		%meta(property="og:title" content="#{yield_content :title}")/
+		%meta(prperty="og:site_name" content="#{yield_content :title}")/
```
Comment (XenHat) [inline views/layout.haml]: this adds metadata to link preview in software such as Telegram Messenger

File: views/layout.haml [hunk 3]
```diff
@@ -11,7 +13,11 @@
 				%ul
 					%li#brand= yield_content :title
 					- if type == :show
-						%li#extension /#{@data[:id]} (#{file_ext(@data[:file]).upcase} - #{File.size(@data[:file]).to_s(:human_size)})
+						- if @data[:type] == "image"
+							%li#extension /#{@data[:id]}.#{@extension} | #{@humanrsize} (#{@width} × #{@height}, #{@ratio})
+						- else
+							%li#extension /#{@data[:id]}.#{@extension} | #{@humanrsize}
+
```
Comment (XenHat) [inline views/layout.haml]: this adds metadata to link preview in software such as Telegram Messenger

File: views/show/video.haml [hunk 4]
```diff
@@ -1,3 +1,5 @@
+%meta(property="og:description" content="#{@data[:id]}.#{@extension} - #{@humanrsize}")/
+%meta(property="og:video:url" content="/#{@data[:id]}.#{@extension}")/
```
Comment (XenHat) [inline views/show/video.haml]: this adds metadata to link preview in software such as Telegram Messenger

File: sharex_server.rb [hunk 5]
```diff
@@ -38,13 +44,38 @@
...
+      @ratio = "21:9"
+    when (8.0/5.0)
+      @ratio = "8:5"
+    when (4.0/3.0)
+      @ratio = "4:3"
+    else
+      @ratio = "#{@trueratio.round(5)}:1"
+    end
+  end
+
```
Comment (XenHat) [inline sharex_server.rb]: Hacky rounding to get friendly aspect ratio

File: sharex_server.rb [hunk 6]
```diff
@@ -38,13 +44,38 @@
...
   return @data unless @data.is_a?(Hash)
   status 200
   haml :show, locals: {type: :show, object: @data[:type]}
 end
 
 get "/:id/download/?" do
+  logger.info "Serving #{params[:id].to_s} to #{request.ip}"
```
Comment (XenHat) [inline sharex_server.rb]: More logging is good

File: views/layout.haml [hunk 7]
```diff
@@ -21,4 +27,4 @@
 		%footer
 			%p
 				%span.copy &copy;&nbsp;
-				= settings.copyright
\ No newline at end of file
+				= settings.copyright
```
Comment (XenHat) [inline views/layout.haml]: Kill me.

**Comment by XenHat** [comment]
I`ll clean and split this tomorrow

**Pull request closed without merging**